pkg/services/k8s/apiserver/admission/schema_translate.go

```diff
@@ -5,88 +5,80 @@ import (
 	"encoding/json"
 	"io"
 
-	"github.com/grafana/thema"
 	"github.com/grafana/thema/vmux"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apiserver/pkg/admission"
 
-	"github.com/grafana/grafana/pkg/cuectx"
 	"github.com/grafana/grafana/pkg/infra/log"
-	"github.com/grafana/grafana/pkg/kinds/dashboard"
+	"github.com/grafana/grafana/pkg/registry/corekind"
 )
 
 const PluginNameSchemaTranslate = "TranslateSchema"
 
 // Register registers a plugin
-func RegisterSchemaTranslate(plugins *admission.Plugins) {
+func RegisterSchemaTranslate(plugins *admission.Plugins, reg *corekind.Base) {
```

Comment:
this function is used to register an admission plugin for all k8s objects. instead of passing in a single kind, could the incoming object be checked if it matches the one of the core kinds?

Reply:
yes, that's what this was aiming for - the `corekind.Base` is a registry of all core kinds, and [this line](https://github.com/grafana/grafana/pull/70728/files#diff-63a981284cbee9d9306cc80687fef6435fd6e6479dc1ebcdc73c7b3370fc11a7R36) is where it generically matches one of the core kinds at runtime. A registry that contains all (static) core and (dynamically registered) custom kinds will be able to apply the same logic, once such a registry exists.

It looks like my mistake here was assuming that this `RegisterSchemaTranslate` func was a wire injector, so it would receive this type automatically if i'd rerun `make gen-go`. i'll update the call site.